extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/FastBootHibernateReactivePersistenceProvider.java

```diff
@@ -317,6 +319,15 @@ private void registerVertxAndPool(String persistenceUnitName,
         serviceRegistry.addInitiator(new VertxInstanceInitiator(vertxHandle.get()));
     }
 
+    private static InjectableInstance<Pool> findVertxPool(Optional<String> optNamedDataSourceName) {
+        if (optNamedDataSourceName.isEmpty() || DataSourceUtil.isDefault(optNamedDataSourceName.get())) {
+            return Arc.container().select(Pool.class);
+        }
+
+        Annotation qualifier = new ReactiveDataSource.ReactiveDataSourceLiteral(optNamedDataSourceName.get());
+        return Arc.container().select(Pool.class, qualifier);
+    }
+
```

Comment:
Or... align on what's done in Hibernate ORM?

https://github.com/quarkusio/quarkus/blob/8029cacaca751ea2e46714b716867b3794944f13/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/FastBootHibernatePersistenceProvider.java#L402-L409

See also:

https://github.com/quarkusio/quarkus/blob/3e636225ed76dc6dc6cafd477b3e35db28dc49b7/extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/ReactiveDataSourceUtil.java#L21-L23

Reply:
https://github.com/quarkusio/quarkus/pull/47631/commits/0015b7bbb5df70b5c106b59bddc3da3a5043ea89